core/src/main/java/com/google/android/fhir/db/impl/dao/LocalChangeUtils.kt

```diff
@@ -85,16 +87,30 @@ object LocalChangeUtils {
     }
 
     /**
-     * Merge two JSON patch strings by concatenating their elements into a new JSON array.
+     * Merge two JSON patch strings by concatenating their elements, which should JSON patch
+     * operations, into a new JSON array.
+     * For duplicate operations, the operation from second patch is picked.
      */
     @JvmStatic
     private fun mergePatches(firstPatch: String, secondPatch: String): String {
-        val objectMapper = ObjectMapper()
         // TODO: validate patches are RFC 6902 compliant JSON patches
-        val first = objectMapper.readValue(firstPatch, JsonPatch::class.java)
-        val updater = objectMapper.readerForUpdating(first)
-        val merged = updater.readValue(secondPatch, JsonPatch::class.java)
-        return merged.toString()
+        val first = JSONArray(firstPatch)
+        val second = JSONArray(secondPatch)
+        val mergeMap = HashMap<Pair<String, String>, JSONObject>()
+
+        (0 until first.length()).forEach { index ->
+            val operation = first.optJSONObject(index) ?: JSONObject()
+            val key = operation.optString("op") to operation.optString("path")
+            mergeMap[key] = operation
+        }
```

Comment:
I find this use of `forEach` harder to read than a plain `for` loop:

```suggestion
        for (i in 0..first.length()) {
           ...
        }
```

what do you think?

Reply:
Agree.

Reply:
One thing though range `0..n` includes n but `0 until n` yields from 0 till n-1.